packages/style-engine/class-wp-style-engine.php

```diff
@@ -444,6 +444,11 @@ public function get_block_supports_styles( $block_styles, $options ) {
 			$styles_output['classnames'] = implode( ' ', array_unique( $classnames ) );
 		}
 
+		// Return array of CSS values, if any.
+		if ( ! empty( $css_declarations ) ) {
+			$styles_output['declarations'] = $css_declarations;
+		}
+
```

Comment:
👍 

This will need tests. Feel free to copy the ones over at https://github.com/WordPress/gutenberg/pull/42307 (and close that PR) or approve and merge! 😄

Reply:
For neatness sake, I think it might be good to merge #42307 so that we've got a separate record of the style engine change in isolation?

Reply:
Sorry to double up the work @tellthemachines , I should I have said I was pumping that PR out. 

I was interested to see it in practice first, running against trunk and the unit tests. 😇

Reply:
> For neatness sake, I think it might be good to merge https://github.com/WordPress/gutenberg/pull/42307 so that we've got a separate record of the style engine change in isolation?

Yeah, probably best do that 😄

Reply:
This part should be fixed now! But turns out I need to do a similar operation on the [editor side](https://github.com/WordPress/gutenberg/pull/42085/files#diff-647213ff197a6d85c4a4a3a5b7d9a7c4ebd5d28741e0e8428b4629c26d67d098R153). Is there an equivalent JS method I can use?

Reply:
Ah, yes, of course! At first glance, I _think_ you should be able to use [getCSSRule](https://github.com/WordPress/gutenberg/blob/trunk/packages/style-engine/src/index.ts#L64) in a similar way? Here's where it's called in the global styles output currently: https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-site/src/components/global-styles/use-global-styles-output.js#L225

Reply:
I'm pretty sure [getCSSRules](https://github.com/WordPress/gutenberg/blob/trunk/packages/style-engine/src/index.ts#L64) does something similar.

Reply:
Oh nice, thanks!